File: Data Structures & Algorithms/next-greater-element-i/submission-1.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        Stack<Integer> st = new Stack<>();
        HashMap<Integer,Integer> hm = new HashMap<>();
        int n1 = nums1.length,n2 = nums2.length;
        int[] ans = new int[n1];
        for(int i = n2 - 1; i >= 0; i--)
        {
            if(st.isEmpty()) hm.put(nums2[i],-1);
            else
            {
                while(!st.isEmpty() && nums2[st.peek()] <= nums2[i]) st.pop();
                hm.put(nums2[i],st.isEmpty() ? -1 : nums2[st.peek()]);
            }
            st.push(i);
        }
        for(int i = 0; i < n1; i++) ans[i] = hm.get(nums1[i]);
        return ans;
    }
}

File: Data Structures & Algorithms/merge-strings-alternately/submission-1.cpp
class Solution {
public:
    string mergeAlternately(string w1, string w2) {
        int i = 0;
        int n1 = w1.length();
        int n2 = w2.length();
        string ans = "";
        while(i < n1 && i < n2)
        {
            ans+=w1[i];
            ans+=w2[i];
            i++;
        }
        while(i < n1) ans+=w1[i++];
        while(i < n2) ans+=w2[i++];
        return ans;
    }
};